Preparation twig transition (#914)
contao/docs: docs/manual/layout/templates/_index.en.md

weight: 40
---

> A template is used to output a module, content element, form or other component. In the navigation area "Layout" 

Reviewer: ```suggestion
A template is used to output a module, content element, form or other component. In the [navigation area](../../administration-area/accessing-and-structure-of-the-backend/#the-navigation-area) "Layout" 
```

> under "Templates" the files can be created, stored in folders and edited. These adjustments are update proof.

Reviewer 2: To keep PRs easier to review, don't include any unrelated changes like this. Only change the content and leave any existing formatting as is. If you want to change markdown formatting, create a separate PR please.

Author: Regarding the mentioned paragraph, I wanted to keep the description more general, just because of the new, additional Twig functionality.

Furthermore, I understood your hint and will take this into account in the future.

Reviewer 2: That particular paragraph has no changes in content in your PR. The only change is that you added a line break after `"Layout"`, that's what I mean.

Author: Thank you.
Maybe i do have to check my VSCode Setup regarding: 
"Add line breaks after 80 characters in paragraphs." (contribute.md)

Reviewer 2: Oh does it do that then automatically for any file that you edit? Yeah, I think you need to change that ;). Personally I only added the _ruler_ for `140` characters and I am not using any extension to automatically add line breaks for me, as this is too intrusive.

`.vscode/settings.json`

```json
{
    "editor.rulers": [
        {
            "column": 140
        }
    ]
}
```

I also wanted to change the guideline to 140 characters in general, since we currently have a mix anyway.

Author: Will check.
So 140 will be the official

Author: Changed info in #946

{{% children %}}

{{% notice info %}}
Template changes are not necessary if you only need an additional CSS ID or CSS class. For most Contao components, 
you can enter them in the "Expert settings" section. The corresponding names are taken from the templates and 
displayed in the source code.
{{% /notice %}}

{{% notice note %}}
In [debug mode](/en/system/debug-mode/), the template names in the HTML source code are displayed as comments, 
so you can see which template is being used.
{{% /notice %}}

## Twig support

> {{< version "4.13" >}}

Reviewer: ```suggestion
{{< version "4.12" >}}
```

> The output can, in addition to PHP templates, also be done via Twig Templats. Detailed information about this 

Reviewer 2: ```suggestion
In addition to PHP templates the output can also be generated via Twig Templates. Detailed information about this 
```

> In addition to PHP templates the output can also be generated via Twig Templates. Detailed information about this 

Reviewer: ```suggestion
In addition to PHP templates the output can also be generated via Twig templates. Detailed information about this 
```

weight: 40
---

A template is used to output a module, content element, form or other component. In the navigation area "Layout" 
under "Templates" the files can be created, stored in folders and edited. These adjustments are update proof.

{{% children %}}

{{% notice info %}}
Template changes are not necessary if you only need an additional CSS ID or CSS class. For most Contao components, 
you can enter them in the "Expert settings" section. The corresponding names are taken from the templates and 
displayed in the source code.
{{% /notice %}}

{{% notice note %}}
In [debug mode](/en/system/debug-mode/), the template names in the HTML source code are displayed as comments, 
so you can see which template is being used.
{{% /notice %}}

## Twig support

{{< version "4.13" >}}

The output can, in addition to PHP templates, also be done via Twig Templats. Detailed information about this 
is provided in the "[Developer Documentation](https://docs.contao.org/dev/framework/templates/twig/)".

> 
> {{% taxonomylist context="tags" filter="Template" title="Anleitungen" description=true %}}
> 
> 
> 
> 

Reviewer 2: ```suggestion
```
